### PR TITLE
Audit Indexer Job

### PR DESCRIPTION
**Pull request by shiva-rakshith**

### CODE REVIEW

File: hcx-pipeline-jobs/core/src/main/scala/org/swasth/dp/core/job/BaseJobConfig.scala [hunk 1]
```diff
@@ -51,11 +51,11 @@ class BaseJobConfig(val config: Config, val jobName: String) extends Serializabl
   // Default output configurations
   val enrichedOutputTag: OutputTag[util.Map[String, AnyRef]] = OutputTag[util.Map[String, AnyRef]]("enriched-events")
 
-  val retryOutputTag: OutputTag[String] = OutputTag[String]("retry-events")
-  val retryTopic = config.getString("kafka.retry.topic")
+  val retryOutputTag: OutputTag[util.Map[String, AnyRef]] = OutputTag[util.Map[String, AnyRef]]("retry-events")
```
Comment (maheshkumargangula) [inline hcx-pipeline-jobs/core/src/main/scala/org/swasth/dp/core/job/BaseJobConfig.scala:54]: Why did you change it to a Map?

File: hcx-pipeline-jobs/audit-indexer/src/main/java/org/swasth/dp/auditindexer/task/AuditIndexerStreamTask.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		Config conf = configFilePath.map(path -> ConfigFactory.parseFile(new File(path)).resolve())
+				.getOrElse(() -> ConfigFactory.load("resources/audit-indexer.conf").withFallback(ConfigFactory.systemEnvironment()));
+		AuditIndexerConfig config = new AuditIndexerConfig(conf,"AuditIndexer-Job");
+		FlinkKafkaConnector kafkaConnector = new FlinkKafkaConnector(config);
+		AuditIndexerStreamTask auditIndexerStreamTask = new AuditIndexerStreamTask(config, kafkaConnector);
+		try {
+			auditIndexerStreamTask.process(config);
+		} catch (Exception e) {
+			//TODO Add loggers
+			e.printStackTrace();
```
Comment (maheshkumargangula) [inline hcx-pipeline-jobs/audit-indexer/src/main/java/org/swasth/dp/auditindexer/task/AuditIndexerStreamTask.java:40]: we should not suppress exceptions.

File: hcx-pipeline-jobs/audit-indexer/src/main/java/org/swasth/dp/auditindexer/functions/AuditIndexerProcessFunction.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        catch (IOException e) {
+            logger.error("Error while indexing message :: " + event + " :: " + e.getMessage());
+            e.printStackTrace();
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+            logger.error("Error while processing message :: " + event + " :: " + e.getMessage());
+        }
+    }
```
Comment (maheshkumargangula) [inline hcx-pipeline-jobs/audit-indexer/src/main/java/org/swasth/dp/auditindexer/functions/AuditIndexerProcessFunction.java:60]: We should not suppress the excepitons

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: hcx-pipeline-jobs/audit-indexer/src/main/java/org/swasth/dp/auditindexer/functions/AuditIndexerProcessFunction.java (hunk 3 above)
Comment (shiva-rakshith) [reply hcx-pipeline-jobs/audit-indexer/src/main/java/org/swasth/dp/auditindexer/functions/AuditIndexerProcessFunction.java]: updated exception handling.

File: hcx-pipeline-jobs/audit-indexer/src/main/java/org/swasth/dp/auditindexer/task/AuditIndexerStreamTask.java (hunk 2 above)
Comment (shiva-rakshith) [reply hcx-pipeline-jobs/audit-indexer/src/main/java/org/swasth/dp/auditindexer/task/AuditIndexerStreamTask.java]: updated exception handling.

File: hcx-pipeline-jobs/core/src/main/scala/org/swasth/dp/core/job/BaseJobConfig.scala (hunk 1 above)
Comment (shiva-rakshith) [reply hcx-pipeline-jobs/core/src/main/scala/org/swasth/dp/core/job/BaseJobConfig.scala]: This is incorrectly updated, it should be String only. I corrected it.

**Review by maheshkumargangula — APPROVED** [review]
